packages/lesswrong/components/users/UserNameDeleted.tsx

```diff
@@ -1,7 +1,21 @@
 import React from 'react';
 import { registerComponent, Components } from '../../lib/vulcan-lib';
+import { useHover } from '../common/withHover';
+import { userGetDisplayName, userGetProfileUrl } from '../../lib/collections/users/helpers';
+import { Link } from '../../lib/reactRouterWrapper';
 
-const UserNameDeleted = () => {
+/**
+ * Username for a deleted user. Ordinarily, looks like "[anonymous]" and
+ * provides no info about the user. However, if a nonnull `userShownToMods` is
+ * provided and the current user is an admin, then this will reveal the name on
+ * hover-over and work as a link.
+ */
+const UserNameDeleted = ({userShownToAdmins}: {
+  userShownToAdmins?: UsersMinimumInfo|null
+}) => {
+  if (userShownToAdmins) {
+    return <UserNameDeletedWithAdminHover user={userShownToAdmins}/>
+  }
```

Comment:
This should also check whether the current user is an admin, right?

Reply:
I think this would be redundant since `Users.checkAccess` filters out deleted users for non-admins, ensuring it will be null here. Will put the redundant check in anyways just in case, and to make it more legible what that's about.